### PR TITLE
Re-adds the ability to cook goliath meat in lava

### LINKED ISSUE #22023: you used to be able to cook goliath meat in the lava, but now it just ashes
see title, it was a really cool mechanic

### PR DESCRIPTION
**Pull request by Mindustry**
Fixes #22023 

This is my first PR, apologies if I do it wrong.
:cl:
fix: Goliath meat can be cooked in lava again
/:cl:

### CODE REVIEW

**Comment by fludd12** [comment]
:fire: 🍖 :fire:

**Comment by FantasticFwoosh** [comment]
Because lava is really really hot, the recent microwave changes also let you cook goliath meat in a microwave so really if this doesn't get merged through as additional feature, you can change the text & keep goliath meat as is.

**Comment by RandomMarine** [comment]
You could always cook goliath meat in a microwave, if you enjoy toxins remaining in your meat.

File: code/modules/food_and_drinks/food/snacks/meat.dm [hunk 1]
```diff
@@ -176,6 +176,11 @@
 	list_reagents = list("nutriment" = 3, "toxin" = 5)
 	icon_state = "goliathmeat"
 
+/obj/item/weapon/reagent_containers/food/snacks/meat/slab/goliath/burn()
+	visible_message("\The [src] finishes cooking!")
+	new /obj/item/weapon/reagent_containers/food/snacks/meat/steak/goliath/(src.loc)
```
Comment (Fox-McCloud) [inline code/modules/food_and_drinks/food/snacks/meat.dm]: `src` here is implied and therefore, unnecessary.

**Comment by Mindustry** [comment]
@FantasticFwoosh Goliath steak and goliath meat steak are two different things.

@Fox-McCloud All those things you pointed out should be changed and committed but I'm a retard when it comes to git.

**Comment by KorPhaeron** [comment]
Thanks

**Comment by FantasticFwoosh** [comment]
@Mindustry prior to the microwave heating to 1k commit (i'll get back to you with the pr tag), you could OUTRIGHT not cook goliath meat in a microwave, and only on lava.

I am very certain @RandomMarine is incorrect, as its the reaction of the steak catching fire IN lava which is replicated by the microwave for the same steak product.

**Comment by Mindustry** [comment]
@FantasticFwoosh

Goliath meat in a microwave creates a normal meat steak named "goliath meat steak", with the meat steak taking the name of the meat.

Goliath steak is created by burning goliath meat in lava and it produces "goliath steak", unlike the meat steak though, this has no plate, creates no rubbish, can not catch fire, and does not burn in lava.

Though I think you're right that Random is wrong in that a quick look at the contents of goliath meat steak show there are no toxins like there should be but this was only a quick glance and I'm not 100% certain.

**Comment by RandomMarine** [comment]
1. Spawn goliath meat
2. Cook in microwave
3. Check reagent vars/reagent scan
4. See that it inherited the toxins present in basic goliath meat.

**Comment by FantasticFwoosh** [comment]
@Mindustry But the fundamental cooking process is the same but applied in a different way, its a steak of the meat but still essentially goliath meat cooked in heat. Its only circumstantially a steak because that's what putting meat in the microwave does to it, if you cooked it over a bonfire (which also sets it on fire like lava, raising the temp to somewhere between 1k) you would get goliath meat.

> Prior to microwave heat pr  #21774 goliath meat would not cook in a microwave and was exclusive to lava, now the microwave DOES cook goliath steak because it replicates the heat without burning it with the fire, but the meat slab product is automatically converted because food-code into plate-goliath steak.

> All toxic meats (spider meat) remains toxic after cooking, i dont understand the OP's assumption that cooking food renders it safe.

**Pull request merged**